File: front/src/ts/data/Address.ts
import Location from "./Location";

class Address {
    constructor(town: Location, street: string | null, zipCode: string | null) {
        street !== null && street.length === 0 ? this.street = null : this.street = street
        zipCode !== null && zipCode.length === 0 ? this.zipCode = null : this.zipCode = zipCode
        this.town = town
    }

    street: string | null
    zipCode: string | null
    town: Location

    public validate() {
        let valid = true

        if (this.zipCode !== null && this.zipCode.length < 9) valid = false

        return valid
    }
}

export default Address

File: back/src/main/java/com/example/back/data/Address.java
package com.example.back.data;

public class Address {
    private String street; //Поле может быть null
    private String zipCode; //Длина строки должна быть не меньше 9, Поле может быть null
    private Location town; //Поле не может быть null

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public String getZipCode() {
        return zipCode;
    }

    public void setZipCode(String zipCode) {
        this.zipCode = zipCode;
    }

    public Location getTown() {
        return town;
    }

    public void setTown(Location town) {
        this.town = town;
    }
}